### PR TITLE
VSCode plugin + CLI watch

### PR DESCRIPTION
**Pull request by danielgerlag**
# Description

Added Drasi explorer to VSCode plugin that enables the user to manage the drasi instance of the current context.
image

This includes inspecting and deleting Drasi resources (Sources, Queries, Reactions).
You can also attach to a running query to see the live results.

Added Debug and Apply actions to Drasi resources that are discovered in thew editor:
image

When the results windows of the debug session is closed or the connection is broken, the query, indexes, etc... will be cleaned up on the query host.

image

Added `watch` command to the CLI the exposes the same functionality to attach to running queries.
image

These features are supported by the `debug` and `watch` routes added to the Management API.

Also includes 2 small fixes:
- Error reporting for the Debezium reactivator, when it fails.
- The CLI looks for an open port to use for port forwarding, rather than a single hard coded value. This enables multiple sessions concurrently.

## Type of change

<!--

Please select **one** of the following options that describes your change and delete the others. Clearly identifying the type of change you are making will help us review your PR faster, and is used in authoring release notes.

If you are making a bug fix or functionality change to Drasi and do not have an associated issue link please create one now. 

- This pull request adds or changes features of Drasi and has an approved issue (issue link required).

### CODE REVIEW

File: dev-tools/vscode/drasi/src/test/runTest.ts [hunk 1]
```diff
@@ -29,7 +29,13 @@ async function main() {
 		const extensionTestsPath = path.resolve(__dirname, './suite/index');
 
 		// Download VS Code, unzip it and run the integration test
-		await runTests({ extensionDevelopmentPath, extensionTestsPath });
+		await runTests({ 
+			extensionDevelopmentPath,
```
Comment (ruokun-niu) [inline dev-tools/vscode/drasi/src/test/runTest.ts:33]: wow :astonished: tests for the vscode plugin

File: dev-tools/vscode/drasi/resources/radioactive.svg
Comment (ruokun-niu) [inline dev-tools/vscode/drasi/resources/radioactive.svg:1]: rip radioactive icon

File: dev-tools/vscode/drasi/src/codelens-provider.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    private drasiClient: DrasiClient;
+
+    constructor(extensionUri: vscode.Uri, drasiClient: DrasiClient) {
+        this.extensionUri = extensionUri;
+        this.drasiClient = drasiClient;
+        vscode.commands.registerCommand('editor.query.run', this.runQuery.bind(this));
+        vscode.commands.registerCommand('editor.resource.apply', this.applyResource.bind(this));
+    }
+
+    provideCodeLenses(document: vscode.TextDocument, token: vscode.CancellationToken): vscode.CodeLens[] {
```
Comment (ruokun-niu) [inline dev-tools/vscode/drasi/src/codelens-provider.ts:19]: do we need codelenses for reactionproviders and sourceproviders?

**Review by ruokun-niu — APPROVED** [review]

File: dev-tools/vscode/drasi/src/codelens-provider.ts (hunk 2 above)
Comment (danielgerlag) [reply dev-tools/vscode/drasi/src/codelens-provider.ts:19]: Let's look into that for the next iteration

**Review by agentofreality — APPROVED** [review]

**Pull request merged**